rpc/gnmi/mixed-schema.md

```diff
@@ -44,14 +41,14 @@ within the message is uniquely identified by the tuple of `<origin, path>`.
 The `origin` field is valid in any context of a `Path` message. Typically it is
 used:
 
- * In a `SetRequest` to indicate a particular schema is being used to modify
-   the target configuration.
- * In a `GetRequest` to retrieve the contents of a particular schema, or in
-   a `GetResponse` to indicate that the payload contains data from a
-   particular `<origin, path>` schema.
- * In a `SubscribeRequest` to subscribe to paths within a particular schema,
-   or `SubscribeResponse` to indicate an update corresponds to a particular
-   `<origin, path>` tuple.
+* In a `SetRequest` to indicate a particular schema is being used to modify
+  the target configuration.
```

Comment:
@earies please suggest the relevant schemas and encodings for JunOS (set request, curly brace and junos yang) be indicated in the origin and encoding.

Reply:
@earies - I'm not sure that I follow your comment here. Origin `cli` was defined to mean "some out of band agreed ASCII format that the device supports". In this case, for JUNOS it could be either the `set xxx` or your `{}` format -- but isn't that something that you're agreeing outside of the concept of this "special" origin?

Maybe the concern here is the word `schema` in the sentence above? Should we rather use `tree`? I think of the different origins as different trees that define some configuration / telemetry representation on the device.

Reply:
This seems super-complicated to me -- especially concepts like `cli_set_val`.

I'd suggest we come back to what we feel is reasonable to support given existing use cases that are being proposed. Otherwise, we're going to end up trying to figure out how to support many different combinations that there will be no real-use-case test coverage for because they are not operationally interesting.

I've documented a simpler proposal that I think gives us some room to simplify below.

Reply:
It's actually simpler than it looks in the previous text.  Let's put aside the concept of `cli_set_val` and `xml_val` for now since I wanted to focus on how an `origin` and `encoding` work in tandem so we have a better contract and understanding of what origin is meant to entail

If we define an `origin` as a schema-set or tree - we have a clear understanding of what this means - it is a **target tree**.  The current method of `cli` does not mean this - it means a blob of ascii text (tied at the hip to the encoding + which by description means _only_ ASCII - no utf8 which can also be supported) that is left up to interpretation and implementation as to what schema trees can be represented.  If by chance a vendor CLI exposes 3rd party modeling, it could entail this for some and not others...

The reason for the suggested change from `cli` -> `native` here is to accommodate above and also bring this back to what the `openconfig` origin already is.  You now have a nice divide of the tree + encoding.

The reason for the proposed addition of `cli_val` is that this too would be a string and still agreed upon out of band as to "which style" of CLI is the supported syntax.  In the case of JUNOS, this is where we would prefer the `{}` curly brace variant.  Other representations could be specified by precise `TypedValue`'s to further define the contract.

Similarly - in JUNOS we have had concepts for years at the API layer to specify encodings and CLI style variants.

https://www.juniper.net/documentation/us/en/software/junos/junos-xml-protocol/topics/ref/tag/junos-xml-protocol-load-configuration.html

The looser the API or if we are overloading fields, origins, etc... the more issues on interpretation we are going to have and this has already been occurring in many ways to date